### ISSUE TITLE
[Category] Short description

### ISSUE THREAD

**Issue by 0xdeadc0d3** [sp614x/optifine#7649]
## Description of Issue

Minecraft freezes on the loading screen after changing game mode/lobby on Hypixel for several times. This causes me to get disconnected from the lobby/game mode. This bug occurs when the render distance is set to 12. If I set it to 8, the freeze is a bit shorter and I can at least play.

Video: https://youtu.be/yYdQ0gAahh0

## Steps to Reproduce

1. Set render distance to 12
2. Connect to mc.hypixel.com
3. Change lobby several times (usually 2-6)

## OptiFine Version

OptiFine HD U G5 (also reproducible with G6 pre1) for Minecraft 1.12.2

## Installation Method

Standalone installer

## Log Files/Crash Reports

https://pastebin.com/vvxuBkuj

I get these exceptions when changing game mode/lobby on Hypixel. But I think the exceptions are probably not related to the issue as I see the exact same exceptions on vanilla (linux) and Windows 10 x64

## F3 Debug Screenshot

![Screenshot from 2024-03-30 01-18-42](https://github.com/sp614x/optifine/assets/85187695/26ad20b8-fdc5-4de3-992b-2dde0cfc1f73)

## Prior Testing

- Does not happen in Vanilla without OptiFine
- Does not happen on Windows 10 with Optifine
- Allocating more memory does not help
- Removing .minecraft folder does not help
- Reinstalling minecraft-launcher does not help

## Additional Information

- DE: GNOME 46.0
- WM: Mutter

**Comment by LimesKey** [comment]
I believe I'm experiencing the same issue, however, in my instance, the freeze never recovers and instead Minecraft times out and crashes itself. I'm on Linux Mint, Minecraft 1.8.9 and OptiFine is my only mod.
```
[21:30:09] [Client thread/INFO]: [CHAT]        
[21:30:09] [Netty Epoll Client IO #0/ERROR]: NetworkDispatcher exception
io.netty.handler.timeout.ReadTimeoutException
[21:30:09] [Client thread/INFO]: Stopping!
[21:30:09] [Client thread/INFO]: SoundSystem shutting down...
[21:30:09] [Client thread/WARN]: Author: Paul Lamb, www.paulscode.com
```

**Comment by 0xdeadc0d3** [comment]
@LimesKey I recommend you install the "Patcher" mod. It fixes the issue

**Comment by LimesKey** [comment]
[This](https://sk1er.club/mods/patcher) mod? It did not fix the issue for me, just to confirm, without this mod the issue comes back?

**Comment by 0xdeadc0d3** [comment]
> without this mod the issue comes back?

Yes. With this mod world switching is almost instant, and the game doesn't crash
